File: app/src/main/java/com/countrydelight/lintruleproject/java_test/FunctionMaxLengthTest.java
package com.countrydelight.lintruleproject.java_test;

public class FunctionMaxLengthTest {

    // right function with max length
    public String rightFunctionMaxLength() {
        return "Hello";
    }

    // wrong fun max length
    public void wrongFunMaxLength() {
        int a = 1;
        int b = 2;
        int c = 3;
        int d = 4;
        int e = 5;
        int f = 6;
        int g = 7;
        int h = 8;
        int i = 9;
        int j = 10;
        int k = 11;
        int l = 12;
        int m = 13;
        int n = 14;
        int o = 15;
        int p = 16;
        int q = 17;
        int r = 18;
        int s = 19;
        int t = 20;
        int u = 21;
        int v = 22;
        int w = 23;
        int x = 24;
        int y = 25;
        int z = 26;

        for (int i1 = 0; i1 < 10; i1++) {
            a += i1;
            b -= i1;
            c *= i1;
            d /= (i1 + 1);
            e += i1;
            f -= i1;
            g *= i1;
            h /= (i1 + 1);
            i += i1;
            j -= i1;
            k *= i1;
            l /= (i1 + 1);
            m += i1;
            n -= i1;
            o *= i1;
            p /= (i1 + 1);
            q += i1;
            r -= i1;
            s *= i1;
            t /= (i1 + 1);
            u += i1;
            v -= i1;
            w *= i1;
            x /= (i1 + 1);
            y += i1;
            z -= i1;
        }

        if (a > 10) {
            a = 0;
        }

        switch (b) {
            case 1:
                a++;
                break;
            case 2:
                a--;
                break;
            case 3:
                a *= 2;
                break;
            default:
                a /= 2;
                break;
        }

        while (c < 100) {
            c += 5;
        }

        try {
            int result = a / (b + 1);
            System.out.println("Result: " + result);
        } catch (ArithmeticException exception) {
            System.out.println("Cannot divide by zero");
        }

        String[] array = new String[10];
        for (int i2 = 0; i2 < array.length; i2++) {
            array[i2] = "Element " + i2;
        }

        for (String element : array) {
            System.out.println(element);
        }

        int sum = 0;
        for (int i3 = 1; i3 <= 100; i3++) {
            sum += i3;
        }

        System.out.println("Sum: " + sum);
    }
}

File: app/src/main/java/com/countrydelight/lintruleproject/kotlin_test/FunctionMaxLengthTest.kt
package com.countrydelight.lintruleproject.kotlin_test

//right function with max length
fun rightFunctionMaxLength(): String {
    return "Hello World"
}

// wrong fun max length
fun wrongFunMaxLength() {
    var a = 1
    var b = 2
    var c = 3
    var d = 4
    var e = 5
    var f = 6
    var g = 7
    var h = 8
    var i = 9
    var j = 10
    var k = 11
    var l = 12
    var m = 13
    var n = 14
    var o = 15
    var p = 16
    var q = 17
    var r = 18
    var s = 19
    var t = 20
    var u = 21
    var v = 22
    var w = 23
    var x = 24
    var y = 25
    var z = 26

    for (i1 in 0..9) {
        a += i1
        b -= i1
        c *= i1
        d /= (i1 + 1)
        e += i1
        f -= i1
        g *= i1
        h /= (i1 + 1)
        i += i1
        j -= i1
        k *= i1
        l /= (i1 + 1)
        m += i1
        n -= i1
        o *= i1
        p /= (i1 + 1)
        q += i1
        r -= i1
        s *= i1
        t /= (i1 + 1)
        u += i1
        v -= i1
        w *= i1
        x /= (i1 + 1)
        y += i1
        z -= i1
    }

    if (a > 10) {
        a = 0
    }

    when (b) {
        1 -> a++
        2 -> a--
        3 -> a *= 2
        else -> a /= 2
    }
    while (c < 100) {
        c += 5
    }

    try {
        val result = a / (b + 1)
        println("Result: $result")
    } catch (e: ArithmeticException) {
        println("Cannot divide by zero")
    }

    val array = arrayOfNulls<String>(10)
    for (i2 in array.indices) {
        array[i2] = "Element $i2"
    }

    for (element in array) {
        println(element)
    }

    var sum = 0
    for (i3 in 1..100) {
        sum += i3
    }

    sum += a
    sum += b
    sum += c
    sum += d
    sum += e
    sum += f
    sum += g
    sum += h
    sum += i
    sum += j
    println("Sum: $sum")
}